vllm/platforms/__init__.py

```diff
@@ -1,123 +1,48 @@
-    is_cpu = "cpu" in version("vllm")
-except Exception:
-    pass
-
-is_neuron = False
-try:
-    import transformers_neuronx  # noqa: F401
-    is_neuron = True
-except ImportError:
-    pass
-
-is_openvino = False
-try:
-    from importlib.metadata import version
-    is_openvino = "openvino" in version("vllm")
-except Exception:
-    pass
-
-if is_tpu:
-    # people might install pytorch built with cuda but run on tpu
-    # so we need to check tpu first
-    from .tpu import TpuPlatform
-    current_platform = TpuPlatform()
-elif is_cuda:
-    from .cuda import CudaPlatform
-    current_platform = CudaPlatform()
-elif is_rocm:
-    from .rocm import RocmPlatform
-    current_platform = RocmPlatform()
-elif is_hpu:
-    from .hpu import HpuPlatform
-    current_platform = HpuPlatform()
-elif is_xpu:
-    from .xpu import XPUPlatform
-    current_platform = XPUPlatform()
-elif is_cpu:
-    from .cpu import CpuPlatform
-    current_platform = CpuPlatform()
-elif is_neuron:
-    from .neuron import NeuronPlatform
-    current_platform = NeuronPlatform()
-elif is_openvino:
-    from .openvino import OpenVinoPlatform
-    current_platform = OpenVinoPlatform()
-else:
-    current_platform = UnspecifiedPlatform()
+from .registry import PlatformRegistry, detect_current_platform
+
+
+class CurrentPlatform:
+    """A wrapper that provides an interface to the current platform.
+    
+    `current_platform` is imported to many modules once vLLM is imported.
+    Updating `current_platform` value directly will not work in those modules.
+    So it needs the wrapper here to provide a dynamic platform loading
+    mechanism.
+
+    This class can make sure that the `current_platform` is always up-to-date.
+    """
+
+    def __init__(self):
+        self.platform: Platform = UnspecifiedPlatform()
+
+    def initialize_current_platform(self):
+        """Initialize the current platform. This function is called when loading
+        the vllm plugin."""
+        # Get the current platform from the registry first. If the current
+        # platform is not set, try to detect the current platform.
+        if PlatformRegistry.current_platform is not None:
+            self.platform = PlatformRegistry.get_current_platform_cls()
+        else:
+            self.platform = detect_current_platform()
+
+        # Register custom ops for the current platform.
+        from vllm.attention.layer import register_custom_ops
+        register_custom_ops()
+
+    def __getattr__(self, name):
+        """Get the attribute. If the attribute is not found, go pass to the 
+        current platform."""
+        if name == 'platform':
+            return self.platform
+        if name == 'initialize_current_platform':
+            return self.initialize_current_platform
+        # Go pass to the current platform.
+        return getattr(self.platform, name)
+
+
+# The global variable for other modules to use.
+current_platform: CurrentPlatform = CurrentPlatform()
```

Comment:
```suggestion
current_platform = CurrentPlatform()
```
Unnecessary type annotation

Reply:
Done